### PR TITLE
Add rhel7.9 support for a100

### PR DESCRIPTION
**Pull request by sharsonia**

### CODE REVIEW

**Review by squirrelsc — DISMISSED** [review]

**Comment by squirrelsc** [comment]
Please make sure to support V3 also.

**Comment by sharsonia** [comment]
> Please make sure to support V3 also.

Yes once I see the test passing consistently, i will push a PR for v3 too.

**Review by squirrelsc — DISMISSED** [review]

**Comment by sharsonia** [comment]
Test is passing with all these changes

https://linuxpipeline.westus2.cloudapp.azure.com/job/Azure/job/job-azure-cloudtest-manual/17349

image

File: Testscripts/Linux/gpu-driver-install.sh [hunk 1]
```diff
@@ -86,20 +86,26 @@ function InstallCUDADrivers() {
...
 
-		rpm -ivh /tmp/"$CUDA_REPO_PKG"
-		LogMsg "Installed the rpm package, $CUDA_REPO_PKG"
+		rpm -ivh /tmp/"${CUDA_REPO_PKG}"
+		LogMsg "Installed the rpm package, ${CUDA_REPO_PKG}"
+
+		# For RHEL/CentOS, it might be needed to install vulkan-filesystem to install CUDA drivers.
+		# Download and Install vulkan-filesystem
+		wget http://mirror.centos.org/centos/7/os/x86_64/Packages/vulkan-filesystem-1.1.97.0-1.el7.noarch.rpm
+		yum -y install vulkan-filesystem-1.1.97.0-1.el7.noarch.rpm
```
Comment (LiliDeng) [inline Testscripts/Linux/gpu-driver-install.sh]: please make sure centos7.x and centos8.x work well.

File: Testscripts/Linux/gpu-driver-install.sh [hunk 2]
```diff
@@ -46,7 +46,7 @@ function skip_test() {
 			if [[ $DISTRO == "redhat_7" ]]; then
 				# RHEL 7.8 should be skipped
 				_minor_ver=$(echo $VERSION_ID | cut -d'.' -f 2)
-				if [ $_minor_ver -gt 7 ]; then
+				if [[ $_minor_ver -eq 8  || $_minor_ver -gt 9 ]]; then
```
Comment (LiliDeng) [inline Testscripts/Linux/gpu-driver-install.sh]: can you add comments for what will be skip, what will be ran here?

File: Testscripts/Linux/utils.sh [hunk 3]
```diff
@@ -2183,7 +2183,10 @@ function apt_get_remove () {
 function update_rhui () {
 	GetDistro
 	if [[ $DISTRO == *"redhat"* ]]; then
-		sudo yum update -y --disablerepo='*' --enablerepo='*microsoft*'
+		# No EUS support for RHEL 7.8 and 7.9
```
Comment (LiliDeng) [inline Testscripts/Linux/utils.sh]: could you please put the doc link here?

File: Testscripts/Windows/GPU-DRIVER-INSTALL.ps1 [hunk 4]
```diff
@@ -221,42 +221,50 @@ function Main {
...
-                    Write-LogErr "Unable to install the LIS RPMs!"
-                    $currentTestResult.TestResult = Get-FinalResultHeader -resultarr "ABORTED"
-                    return $currentTestResult
-                }
-                # Restart VM to load the new LIS drivers
-                if (-not $TestProvider.RestartAllDeployments($allVMData)) {
-                    Write-LogErr "Unable to connect to VM after restart!"
-                    $currentTestResult.TestResult = Get-FinalResultHeader -resultarr "ABORTED"
-                    return $currentTestResult
+            if ((($majorVersion -eq 7) -and ($minorVersion -lt 8)) -or $majorVersion -lt 7) {
```
Comment (LiliDeng) [inline Testscripts/Windows/GPU-DRIVER-INSTALL.ps1:231]: add comments here to list supported/not supported distro, thanks

File: XML/Other/ReplaceableTestParameters.xml [hunk 5]
```diff
@@ -614,7 +614,7 @@ Scenario 2 : You need to run all the performance tests quickly.
 	<Parameter>
 		<ReplaceThis>CUDA_DRIVER</ReplaceThis>
 		<!-- Reference: https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html -->
-		<ReplaceWith>10.1.105-1</ReplaceWith>
+		<ReplaceWith>10.0.130-1</ReplaceWith>
```
Comment (LiliDeng) [inline XML/Other/ReplaceableTestParameters.xml:617]: not sure which version is the proper one, but use latest one would be better? if have to use this one, make sure no regression on other distro

Comment (sharsonia) [reply XML/Other/ReplaceableTestParameters.xml:617]: Recommended in Azure Docs

https://docs.microsoft.com/en-us/azure/virtual-machines/linux/n-series-driver-setup

image

File: Testscripts/Linux/gpu-driver-install.sh (hunk 1 above)
Comment (sharsonia) [reply Testscripts/Linux/gpu-driver-install.sh]: Both rhel and centos > 7.x requires vulcan-filesystem for CUDA installation

File: Testscripts/Windows/GPU-DRIVER-INSTALL.ps1 [hunk 6]
```diff
@@ -221,42 +221,51 @@ function Main {
...
-                    Write-LogErr "Unable to install the LIS RPMs!"
-                    $currentTestResult.TestResult = Get-FinalResultHeader -resultarr "ABORTED"
-                    return $currentTestResult
-                }
-                # Restart VM to load the new LIS drivers
-                if (-not $TestProvider.RestartAllDeployments($allVMData)) {
-                    Write-LogErr "Unable to connect to VM after restart!"
-                    $currentTestResult.TestResult = Get-FinalResultHeader -resultarr "ABORTED"
-                    return $currentTestResult
+            # LIS drivers are only requiredfor RHEL/CentOS versions < 7.8
```
Comment (LiliDeng) [inline Testscripts/Windows/GPU-DRIVER-INSTALL.ps1]: required for

**Review by LiliDeng — DISMISSED** [review]

**Review by LiliDeng — DISMISSED** [review]

**Review by LiliDeng — DISMISSED** [review]

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**